ahnlich/ai/src/engine/ai/providers/ort.rs

```diff
@@ -0,0 +1,266 @@
+                    .unwrap_or_else(|_| panic!("Failed to retrieve {} ", weights_file));
+                let executioners: Vec<ExecutionProviderDispatch> = Default::default();
+                let session = Session::builder()
+                    .unwrap()
+                    .with_execution_providers(executioners)
+                    .unwrap()
+                    .with_optimization_level(GraphOptimizationLevel::Level3)
+                    .unwrap()
+                    .with_intra_threads(threads)
+                    .unwrap()
+                    .commit_from_file(model_file_reference)
+                    .unwrap();
+                self.model = Some(ORTModel::Image(ORTImageModel {
+                    repo_name,
+                    weights_file,
+                    input_param,
+                    output_param,
+                    session: Some(session),
+                }));
+            }
+        }
+        self
+    }
+
+    fn get_model(&self) {
+        let cache_location = self
+            .cache_location
+            .clone()
+            .expect("Cache location not set.");
+        let supported_model = self.supported_models.expect("A model has not been set.");
+        let ort_model = ORTModel::try_from(&supported_model).unwrap();
+
+        let cache = Cache::new(cache_location);
+        let api = ApiBuilder::from_cache(cache)
+            .with_progress(true)
+            .build()
+            .unwrap();
+
+        match ort_model {
+            ORTModel::Image(ORTImageModel {
+                repo_name,
+                weights_file,
+                ..
+            }) => {
+                let model_repo = api.model(repo_name);
+                model_repo
+                    .get(&weights_file)
+                    .unwrap_or_else(|_| panic!("Failed to retrieve {} ", weights_file));
+                let preprocessor = model_repo.get("preprocessor_config.json");
+                if preprocessor.is_err() {
+                    log::warn!(
+                        "Failed to retrieve preprocessor_config.json for model: {}",
+                        supported_model
+                    );
+                }
+            }
+        }
+    }
+
+    fn run_inference(&self, input: &ModelInput, action_type: &InputAction) -> Result<Vec<f32>, AIProxyError> {
+        if let Some(ORTModel::Image(ORTImageModel {
+            session,
+            input_param,
+            output_param,
+            ..
+        })) = &self.model
+        {
+            let image = match input {
+                ModelInput::Image(image) => image.clone(),
+                _ => {
+                    let store_input_type: AIStoreInputType = input.into();
+                    let index_model_repr: Model = (&self.supported_models.expect(
+                        &AIProxyError::AIModelNotInitialized.to_string())).into();
+                    match action_type {
+                        InputAction::Query => {
+                            panic!(
```

Comment:
There's no need to panic if we are already passing the error upwards... we can just do `return Err(AIProxyError::StoreSetTypeMismatchError(_))`

Reply:
Also `action_type` doesn't really matter here as both variants return the same error

Reply:
The variants are: `StoreQueryTypeMismatchError` and `StoreSetTypeMismatchError`, are they the same?